app/src/main/java/com/amaze/filemanager/database/UtilsHandler.java

```diff
@@ -79,13 +86,22 @@ public void onCreate(SQLiteDatabase db) {
                 + COLUMN_NAME + " TEXT,"
                 + COLUMN_PATH + " TEXT"
                 + ")";
+        String querySftp = "CREATE TABLE IF NOT EXISTS " + TABLE_SFTP + " ("
+                + COLUMN_ID + " INTEGER PRIMARY KEY,"
+                + COLUMN_NAME + " TEXT,"
+                + COLUMN_PATH + " TEXT,"
+                + COLUMN_HOST_PUBKEY + " TEXT,"
+                + COLUMN_PRIVATE_KEY_NAME + " TEXT,"
+                + COLUMN_PRIVATE_KEY + " TEXT"
+                + ")";
 
         db.execSQL(queryHistory);
         db.execSQL(queryHidden);
         db.execSQL(queryList);
         db.execSQL(queryGrid);
         db.execSQL(queryBookmarks);
         db.execSQL(querySmb);
+        db.execSQL(querySftp);
     }
 
     @Override
```

Comment:
Change onUpgrade()'s code to:
```
    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        switch(newVersion) {
            case 2:
                db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SFTP + " ("
                        + COLUMN_ID + " INTEGER PRIMARY KEY,"
                        + COLUMN_NAME + " TEXT,"
                        + COLUMN_PATH + " TEXT,"
                        + COLUMN_HOST_PUBKEY + " TEXT,"
                        + COLUMN_PRIVATE_KEY_NAME + " TEXT,"
                        + COLUMN_PRIVATE_KEY + " TEXT"
                        + ")");
        }
    }
```

And increment the `DATABASE_VERSION` to 2.

Reply:
Why would you change onUpgrade code to that? Consider a change in schema later and you would need to rewrite the code to drop and create those tables again . And creating table in onUpgrade is redundant, simply drop the table if exists and call ````onCreate```` again.

Reply:
That would delete any old data. Why would a schema change make you have to drop the tables? If you don't need some data, sure drop it, but if it is still needed why would you drop it?

Reply:
**Ok, I made a mistake in the code above.**

For version 2 the code would be:
´´´
    @Override
    public void onUpgrade(SQLiteDatabase db, int **oldVersion**, int newVersion) {
        switch(**oldVersion**) {
            case 1:
                db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SFTP + " ("
                        + COLUMN_ID + " INTEGER PRIMARY KEY,"
                        + COLUMN_NAME + " TEXT,"
                        + COLUMN_PATH + " TEXT,"
                        + COLUMN_HOST_PUBKEY + " TEXT,"
                        + COLUMN_PRIVATE_KEY_NAME + " TEXT,"
                        + COLUMN_PRIVATE_KEY + " TEXT"
                        + ")");
        }
    }
´´´
PS: No break, so that if you upgrade from 1 to n you get all changes needed.

Reply:
Alright, what would you suggest then, if I add a new column in a table, what do I do with existing rows? What entries will they contain for that column in their row?

Reply:
Thanks for the explanation and clarifications. See 6a4951e if can suit your needs.

Reply:
@VishalNehra you need to add a new column? 
There are 2 possibilities:
* first, from the information on the other ones (and the device) you'll be able to reconstruct the column for old entries; 
* second, the user must enter the information manually, but, because the user doesn't want to enter 100 values, you must put a special value (on each cell of the new column) and work arround it, you add the new data for new entries.